codecarbon/output.py

```diff
@@ -131,7 +131,17 @@ class BaseOutput(ABC):
     """
 
     @abstractmethod
-    def out(self, data: EmissionsData):
+    def out(self, total: EmissionsData, delta: EmissionsData):
+        pass
+
+    #TODO: document lifecycle scope
+    @abstractmethod
+    def live_out(self, total: EmissionsData, delta: EmissionsData):
```

Comment:
I like that you give `total` AND `delta` to the `live_out`. We could use the total for metrics that are cumulative for instance.